### ISSUE TITLE
Vector3 node crashing on ticking block entity if invalid input is given

### ISSUE THREAD

**Issue by blockninja124** [Low-Drag-MC/Multiblocked2#59]
I was trying to put a direction value (machine front) into the xyz/vector3 node, hoping it would output the vector direction of that direction, but it simply crashed my game as soon as I load a world that has my machine. I looked at the vector threes source, and I think I know why.

```java
@InputPort(name = "xyz")
public Object in;
@OutputPort(name = "xyz")
public Vector3f out;
//...

@Override
public void process() {
    if (in == null) {
        out = internalValue;
    } else if (in instanceof Vector3f vector3f) {
        out = vector3f;
        internalValue = out;
    } //IF in is NOT NULL but also NOT Vector3f, out at this point IS NULL
    out = new Vector3f(
            inX == null ? out.x() : inX, //out.x() crashes, because out is null
            inY == null ? out.y() : inY,
            inZ == null ? out.z() : inZ);
    outX = out.x();
    outY = out.y();
    outZ = out.z();
}
```

Proposed fixes:
1. Change input type of xyz to Vector3f instead of Object
2. add another `else` statement after checking if its a vector3f, and if its not a vector3f make it equal to 0, 0, 0 or something

If fix no. 2 is done, another `if` could check if its a direction object, and turn that into vector form, which could be quite useful.

**Comment by blockninja124** [comment]
I just found the direction info node, so another if statement in the vector3 node probably isn't needed. But still the null crash is pretty annoying

**Comment by Yefancy** [comment]
thanks for your report, will fix it in the next ldlib update

**Comment by Yefancy** [comment]
fixed since ldlib 1.0.33.a
